apollo-router-core/src/query_planner/router_bridge_query_planner.rs

```diff
@@ -91,6 +96,33 @@ mod tests {
         insta::assert_debug_snapshot!(result);
     }
 
+    #[test]
+    fn empty_query_plan() {
+        let expected = PlannerResult::QueryPlan { node: None };
+        let actual: PlannerResult = serde_json::from_value(json!({ "kind": "QueryPlan"})).expect(
+            "If this test fails, It probably means QueryPlan::node isn't an Option anymore.\n
+                 Introspection queries return an empty QueryPlan, so the node field needs to remain optional.",
+        );
+
+        assert_eq!(expected, actual);
```

Comment:
:yarn: Unneeded assertion. Only the unwrapping is relevant here.

```suggestion
        serde_json::from_str::<PlannerResult>(r#"{"kind":"QueryPlan"}"#)
            .expect("introspection queries return an empty QueryPlan");
```

Reply:
This whole test might not be necessary since you test this right after in `empty_query_plan_should_be_a_planner_error`

Reply:
I left it because I'd like to make it explicit that there are two levels of constraints "plan node needs to be optional, and an empty node shall trigger a planner error".

Reply:
fair